src/components/Status/index.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { ReactElement } from 'react'
+import { Icon, Text } from '../index'
+
+import { TStatusProps } from './types'
+import '../../assets/styles/components/_status.scss'
+import classNames from 'classnames'
+
+export const Status = (props: TStatusProps): ReactElement => {
+  const {
+    type = 'primary',
+    size = 'large',
+    text,
+    className = '',
+    leftIconProps,
+    rightIconProps,
+    noIcon
+  } = props
+
+  return (
+    <div className={classNames(`status status--${type} status--${size}`, className)}>
+      {leftIconProps?.name && (
+        <Icon size="xsmall" type={type} {...leftIconProps} className="mr-8" />
+      )}
+      {!noIcon && !leftIconProps?.name ? <span className="status__circle mr-8" /> : null}
+      <Text
+        type={noIcon ? type : 'primary'}
+        size={size == 'large' ? 'standard' : size == 'medium' ? 'small' : 'xsmall'}
```

Comment:
this line code has a complexity, so I think we can do this:

1. Create an enum like this ` 
    `enum StatusTextSize {
       large = 'standard',
      medium = 'small',
      small = 'small'
    `
2. Replace this line with this
    size={StatusTextSize[size]}